pkg/svcat/service-catalog/plan.go

```diff
@@ -55,44 +56,69 @@ type Plan interface {
 	// GetDescription returns the plan description.
 	GetDescription() string
 
+	// GetFree returns if the plan is free.
+	GetFree() bool
+
 	// GetClassID returns the plan's class name.
 	GetClassID() string
+
+	// GetInstanceCreateSchema returns the instance create schema from plan.
+	GetInstanceCreateSchema() *runtime.RawExtension
+
+	// GetInstanceUpdateSchema returns the instance update schema from plan.
+	GetInstanceUpdateSchema() *runtime.RawExtension
+
+	// GetBindingCreateSchema returns the instance create schema from plan.
+	GetBindingCreateSchema() *runtime.RawExtension
 }
 
 // RetrievePlans lists all plans defined in the cluster.
-func (sdk *SDK) RetrievePlans(opts RetrievePlanOptions) ([]Plan, error) {
+func (sdk *SDK) RetrievePlans(classID string, opts ScopeOptions) ([]Plan, error) {
+	plans, err := sdk.retrievePlansByListOptions(opts, metav1.ListOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	if classID == "" {
+		return plans, nil
+	}
+
+	var filtered []Plan
+	for _, p := range plans {
+		if p.GetClassID() == classID {
+			filtered = append(filtered, p)
+		}
+	}
+
+	return filtered, nil
+}
+
+func (sdk *SDK) retrievePlansByListOptions(scopeOpts ScopeOptions, listOpts metav1.ListOptions) ([]Plan, error) {
 	var plans []Plan
 
-	if opts.Scope.Matches(ClusterScope) {
-		csp, err := sdk.ServiceCatalog().ClusterServicePlans().List(v1.ListOptions{})
+	if scopeOpts.Scope.Matches(ClusterScope) {
+		csp, err := sdk.ServiceCatalog().ClusterServicePlans().List(listOpts)
 		if err != nil {
 			return nil, fmt.Errorf("unable to list cluster-scoped plans (%s)", err)
 		}
 
 		for _, p := range csp.Items {
-			if opts.ClassID != "" && p.GetClassID() != opts.ClassID {
-				continue
-			}
-
 			plan := p
```

Comment:
nit:  I realize this is left over from refactored code - - you could remove plan and just use &p on the next line.   Same on line 122.  You can usually append a slice to a slice vs iterating each element like you  are doing here but  the receiver pointer on ServicePlan's GetBindingCreateSchema() prevents that I think.